firebase-firestore/src/commonMain/kotlin/dev/gitlive/firebase/firestore/LocalCacheSettings.kt

```diff
@@ -0,0 +1,90 @@
+package dev.gitlive.firebase.firestore
+
+sealed class LocalCacheSettings {
+
+    internal companion object {
+        // Firestore cache defaults to 100MB
+        const val DEFAULT_CACHE_SIZE = 100L*1024L*1024L
+    }
+
+    data class Persistent(val sizeBytes: Long) : LocalCacheSettings() {
+
+        companion object {
+            fun newBuilder(): Builder = BuilderImpl()
+        }
+
+        interface Builder {
+            var sizeBytes: Long
+            fun build(): Persistent
+        }
+
+        private class BuilderImpl(
+            override var sizeBytes: Long = DEFAULT_CACHE_SIZE
+        ) : Builder {
+            override fun build(): Persistent = Persistent(sizeBytes)
+        }
+    }
+    data class Memory(val garbaseCollectorSettings: GarbageCollectorSettings) : LocalCacheSettings() {
+
+        companion object {
+            fun newBuilder(): Builder = BuilderImpl()
+        }
+
+        interface Builder {
+
+            var gcSettings: GarbageCollectorSettings
```

Comment:
rename garbageCollectorSettings for android sdk binary compat

Reply:
Its called `gcSettings` though: https://firebase.google.com/docs/reference/kotlin/com/google/firebase/firestore/MemoryCacheSettings.Builder

Reply:
sorry yes got confused with the name MemoryCacheSettings where it has a different name 🤯